compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -587,22 +587,22 @@ private void visitAssignment(BLangExpression varRefExpr, TaintedStatus varTainte
             if (varTaintedStatus == TaintedStatus.TAINTED && varRefExpr instanceof BLangVariableReference) {
                 BLangVariableReference varRef = (BLangVariableReference) varRefExpr;
                 if (isMutableVariable(varRef) && isGlobalVarOrServiceVar(varRef) && !isMarkedTainted(varRef)) {
-                    if (varRef.symbol.type.tag == TypeTags.OBJECT) {
-                        addTaintError(pos, varRef.symbol.name.value,
+                    if (varRef.symbol != null && varRef.symbol.type.tag == TypeTags.OBJECT) {
+                        addTaintError(pos, getVariableName(varRef),
                                 DiagnosticCode.TAINTED_VALUE_PASSED_TO_MODULE_OBJECT);
                     } else {
-                        addTaintError(pos, varRef.symbol.name.value,
+                        addTaintError(pos, getVariableName(varRef),
                                 DiagnosticCode.TAINTED_VALUE_PASSED_TO_GLOBAL_VARIABLE);
                     }
                     return;
                 } else if (varRef.symbol != null && varRef.symbol.closure && varTaintedStatus == TaintedStatus.TAINTED
```

Comment:
There's an IDE warning for `varTaintedStatus == TaintedStatus.TAINTED` --> always true.

Shall we fix this too?